java-squid/src/main/java/org/sonar/java/se/SymbolicValue.java

```diff
@@ -257,23 +276,140 @@ public InstanceOfSymbolicValue(int id) {
     }
 
     @Override
-    public ProgramState setConstraint(ProgramState programState, BooleanConstraint booleanConstraint) {
+    public List<ProgramState> setConstraint(ProgramState programState, BooleanConstraint booleanConstraint) {
       if (BooleanConstraint.TRUE.equals(booleanConstraint)) {
         if (NullConstraint.NULL.equals(programState.constraints.get(operand))) {
           // irrealizable constraint : instance of true if operand is null
-          return null;
+          return ImmutableList.of();
         }
         // if instanceof is true then we know for sure that expression is not null.
-        ProgramState ps = operand.setConstraint(programState, NullConstraint.NOT_NULL);
-        if(ps.equals(programState)) {
+        List<ProgramState> ps = operand.setConstraint(programState, NullConstraint.NOT_NULL);
+        if (ps.size() == 1 && ps.get(0).equals(programState)) {
           //FIXME we already know that operand is NOT NULL, so we add a different constraint to distinguish program state. Typed Constraint should store the deduced type.
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Take the required action to fix the issue indicated by this comment. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS1134)